homeassistant/components/sensor/citybikes.py

```diff
@@ -0,0 +1,310 @@
+from homeassistant.helpers.event import (
+    async_track_time_interval, async_track_point_in_utc_time)
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.util import location, distance, dt as dt_util
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_ENDPOINT = 'https://api.citybik.es/{uri}'
+NETWORKS_URI = 'v2/networks'
+STATIONS_URI = 'v2/networks/{uid}?fields=network.stations'
+
+SCAN_INTERVAL = timedelta(minutes=5)  # Timely, and doesn't suffocate the API
+DOMAIN = 'citybikes'
+CONF_NETWORK = 'network'
+CONF_RADIUS = 'radius'
+CONF_STATIONS_LIST = 'stations'
+ATTR_NETWORKS_LIST = 'networks'
+ATTR_NETWORK = 'network'
+ATTR_STATIONS_LIST = 'stations'
+ATTR_ID = 'id'
+ATTR_UID = 'uid'
+ATTR_NAME = 'name'
+ATTR_EXTRA = 'extra'
+ATTR_TIMESTAMP = 'timestamp'
+ATTR_EMPTY_SLOTS = 'empty_slots'
+ATTR_FREE_BIKES = 'free_bikes'
+ATTR_TIMESTAMP = 'timestamp'
+CITYBIKES_ATTRIBUTION = "Information provided by the CityBikes Project "\
+                        "(https://citybik.es/#about)"
+
+
+PLATFORM_SCHEMA = vol.All(
+    cv.has_at_least_one_key(CONF_RADIUS, CONF_STATIONS_LIST),
+    PLATFORM_SCHEMA.extend({
+        vol.Optional(CONF_NAME, default=''): cv.string,
+        vol.Optional(CONF_NETWORK): cv.string,
+        vol.Inclusive(CONF_LATITUDE, 'coordinates'): cv.latitude,
+        vol.Inclusive(CONF_LONGITUDE, 'coordinates'): cv.longitude,
+        vol.Optional(CONF_RADIUS, 'station_filter'): cv.positive_int,
+        vol.Optional(CONF_STATIONS_LIST, 'station_filter'):
+            vol.All(
+                cv.ensure_list,
+                vol.Length(min=1),
+                [cv.string])
+    }))
+
+NETWORK_SCHEMA = vol.Schema({
+    vol.Required(ATTR_ID): cv.string,
+    vol.Required(ATTR_NAME): cv.string,
+    vol.Required(ATTR_LOCATION): vol.Schema({
+        vol.Required(ATTR_LATITUDE): vol.Coerce(float),
+        vol.Required(ATTR_LONGITUDE): vol.Coerce(float),
+        }, extra=vol.REMOVE_EXTRA),
+    }, extra=vol.REMOVE_EXTRA)
+
+NETWORKS_RESPONSE_SCHEMA = vol.Schema({
+    vol.Required(ATTR_NETWORKS_LIST): [NETWORK_SCHEMA],
+    })
+
+STATION_SCHEMA = vol.Schema({
+    vol.Required(ATTR_FREE_BIKES): cv.positive_int,
+    vol.Required(ATTR_EMPTY_SLOTS): cv.positive_int,
+    vol.Required(ATTR_LATITUDE): vol.Coerce(float),
+    vol.Required(ATTR_LONGITUDE): vol.Coerce(float),
+    vol.Required(ATTR_ID): cv.string,
+    vol.Required(ATTR_NAME): cv.string,
+    vol.Required(ATTR_TIMESTAMP): cv.string,
+    vol.Optional(ATTR_EXTRA): vol.Schema({
+        vol.Optional(ATTR_UID): cv.string
+        }, extra=vol.REMOVE_EXTRA)
+    }, extra=vol.REMOVE_EXTRA)
+
+STATIONS_RESPONSE_SCHEMA = vol.Schema({
+    vol.Required(ATTR_NETWORK): vol.Schema({
+        vol.Required(ATTR_STATIONS_LIST): [STATION_SCHEMA]
+        }, extra=vol.REMOVE_EXTRA)
+    })
+
+
+MONITORED_NETWORKS = {}
```

Comment:
Please do not use global variables. You can store data in `hass.data`, a dictionary for exactly this purpose.